src/Plugin/Block/PageHeaderBlock.php

```diff
@@ -0,0 +1,230 @@
+          $entity = $this->currentRouteMatch->getParameter($name);
+          if ($entity instanceof EntityInterface && $entity->hasLinkTemplate('canonical')) {
+            $this->entity = $entity;
+          }
+          // Since entity was found, no need to iterate further.
+          break;
+        }
+      }
+    }
+
+    // Dispatch event to allow modules to alter block content.
+    $event = new PageHeaderDisplayEvent($this->entity);
+    $this->eventDispatcher->dispatch(PageHeaderDisplayEvent::EVENT_NAME, $event);
+
+    // Set the title, lede and visibility.
+    $this->title = is_null($event->getTitle()) ? $this->getTitle() : $event->getTitle();
+    $this->lede = is_null($event->getLede()) ? $this->getLede() : $event->getLede();
+    $this->visible = $event->getVisibility();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $build = [];
+
+    $build[] = [
+      '#theme' => 'localgov_page_header_block',
+      '#title' => $this->title,
+      '#lede' => $this->lede,
+    ];
+
+    return $build;
+  }
+
+  /**
+   * Get the default title for the page.
+   *
+   * @return array|string|null
+   *   Returns a title for the current page or NULL if it can't be determined.
+   */
+  protected function getTitle() {
+    $request = $this->requestStack->getCurrentRequest();
+    $route = $this->currentRouteMatch->getRouteObject();
+
+    return $this->titleResolver->getTitle($request, $route);
+  }
+
+  /**
+   * Get the default lede for page.
+   *
+   * @return array|string|null
+   *   Returns a lede for the current page or NULL if it can't be determined.
+   */
+  protected function getLede() {
+
+    // Return node summary if it exists.
+    if ($this->entity instanceof Node && $this->entity->get('body')) {
+      return [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => $this->entity->get('body')->summary,
+      ];
+    }
+
+    // Return a term string.
+    if ($this->entity instanceof Term) {
+      return [
+        '#type' => 'html_tag',
+        '#tag' => 'p',
+        '#value' => $this->t('All pages relating to @label.', ['@label' => strtolower($this->entity->label())]),
+      ];
+    }
+
+    return NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function blockAccess(AccountInterface $account) {
+    if ($this->visible) {
+      return AccessResult::allowed();
+    }
+    else {
+      return AccessResult::neutral();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return ['label_display' => FALSE];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    if (!is_null($this->entity)) {
+      return Cache::mergeTags(parent::getCacheTags(), $this->entity->getCacheTags());
+    }
+    return parent::getCacheTags();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['route']);
```

Comment:
+1 for switching from **url.path** to **route**.  We have run into a bizarre edge case recently that was caused by url.path.  It was not really a bug, more of a peculiarity of the hosting environment.  But **route** survived fine where url.path didn't.